examples/llm/src/models/mosaic_gpt/mosaic_gpt.py

```diff
@@ -242,6 +273,19 @@ def forward(
                 'prefix_mask is a required argument when MosaicGPT is configured with prefix_lm=True.'
             )
 
+        if self.training:
```

Comment:
[Abhi mention that our eval set is also preconcat and should use this.](https://mosaicml.slack.com/archives/C05000N6VD3/p1680129645540909)
(Sorry I know I'm the one who lead you astray on this one)

Reply:
I think we *have* to skip the input checking outside train mode. I'm fine with continuing to require `sequence_id` being supplied during training when `attn_uses_sequence_id==True`, but requiring it during eval mode is a huge burden.
Even though the input checking below this is skipped if the model is in eval mode, `sequence_id` will still be used if supplied (and the flag is turned on). That is, it'll handle our eval dataset correctly. So, I feel like this is a good compromise.

Reply:
I guess if eval ds passes `sequence_id` to fwd, then it is used right?